### PR TITLE
[CP-20164] Allow users to set secret filepath and filename for agent server container

### PR DESCRIPTION
**Pull request by dmepham**
By submitting a PR to this repository, you agree to the terms within the [CloudZero Code of Conduct](https://github.com/cloudzero/template-cloudzero-open-source/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/cloudzero/template-cloudzero-open-source/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

users may want to manage secrets with Hashicrop Vault, which requires no k8s Secrets at all. instead, it injects secrets to the container filesystem based on an annotation. the filename and filepath that the secret is inject to is based on the annotation provided by the user. to allow users to use Vault, we need to allow arbitrary secret filepaths to be specified, and we need to allow users to deploy without any k8s secret at all

### References
Vault documentation: https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-sidecar

### Testing

1. Assert that `apiKey` method continues to work as expected:
```bash
helm upgrade --install cloudzero-agent ./ --set apiKey=$api_key --set clusterName=danm --set region=us-east-1 --set-string cloudAccountId=******** --set kube-state-metrics.enabled=true --set prometheus-node-exporter.enabled=true
```
- confirmed that secret is created, writes are successful

2. Assert that `existingSecretName` method continues to work as expected:
```bash
helm upgrade --install cloudzero-agent ./ --set existingSecretName=cloudzero-agent-key-existing --set clusterName=danm --set region=us-east-1 --set-string cloudAccountId=975482786146 --set host=dev-api.cloudzero.com --set kube-state-metrics.enabled=true --set prometheus-node-exporter.enabled=true
```
- confirmed that secret is NOT created, writes are successful

3. Assert that we can use hashicorp vault to get the api key, and use `containerSecretFilePath` and `containerSecretFileName` to get the vault-injected key.
I'm following [this doc](https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-sidecar#install-the-vault-helm-chart) and adjusting for use with our chart, and omitting some steps for brevity

**install development vault server**
```bash
helm repo add hashicorp https://helm.releases.hashicorp.com
helm repo update
helm install vault hashicorp/vault
```
**set secret in vault**
```bash
kubectl exec -it vault-0 -- /bin/sh
$ vault kv put internal/test/cz-api-key apikey=********
```

**deploy the helm chart**

Given the file test.yaml:
```yaml
server:
  podAnnotations:
    vault.hashicorp.com/agent-inject: 'true'
    vault.hashicorp.com/role: 'cloudzero-agent-with-ui-server'
    vault.hashicorp.com/agent-init-first: 'true'
    vault.hashicorp.com/agent-inject-secret-my-special-key-filename: 'internal/test/cz-api-key'
    vault.hashicorp.com/agent-inject-template-my-special-key-filename: |
      {{- with secret "internal/data/test/cz-api-key" -}}
      {{ .Data.data.apikey }}
      {{- end -}}
```
install with the following command:

```bash
helm upgrade --install cloudzero-agent ./  --set clusterName=danm --set region=us-east-1 --set-string cloudAccountId=12345678910 --set kube-state-metrics.enabled=true --set prometheus-node-exporter.enabled=true  -f test.yaml --set server.containerSecretFilePath=/vault/secrets/ --set server.containerSecretFileName=my-special-key-filename
```
Confirmed that no secret is created, the prometheus agent uses the vault-injected key, and remote writes are successful

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

**Review by bdrennz — APPROVED** [review]
Looks good!

**Review by josephbarnett — CHANGES_REQUESTED** [review]
I think we need a small change to the git workflow for installation test:
> https://github.com/Cloudzero/cloudzero-charts/actions/runs/9984162965/job/27592934253?pr=63#step:8:490

Otherwise this looks good

File: charts/cloudzero-agent/values.yaml [hunk 1]
```diff
@@ -93,6 +92,10 @@ server:
   - --web.console.libraries=/etc/prometheus/console_libraries
   - --web.console.templates=/etc/prometheus/consoles
   - --enable-feature=agent
+  # -- The agent will use this file path on the container filesystem to get the CZ API key.
+  containerSecretFilePath: /etc/config/prometheus/secrets/
+  # -- The agent will look for a file with this name to get the CZ API key.
+  containerSecretFileName: value
```
Comment (dmepham) [inline charts/cloudzero-agent/values.yaml:98]: I originally named with `cz-api-key`, which I think is a better name than `value`. but I'm worried that users that have created secrets externally with the `value` name could see this fail on upgrade

**Review by josephbarnett — APPROVED** [review]
👍

**Pull request merged**